### PR TITLE
[dev] make utils::{shr, shl} generic over input types

### PR DESCRIPTION
**Pull request by TAdev0**
- [x] issue #46 
- [x] follows contribution [guide](https://github.com/keep-starknet-strange/raito/blob/main/CONTRIBUTING.md)
- [x] code change includes tests

### CODE REVIEW

File: src/utils.cairo [hunk 1]
```diff
@@ -1,34 +1,90 @@
...
-    value * fast_pow(2.into(), shift.into())
+pub trait Bitshift<T, U> {
+    fn shl(self: T, shift: U) -> T;
+    fn shr(self: T, shift: U) -> T;
 }
 
-// Bitwise shift right for u256
-pub fn shr(value: u256, shift: u32) -> u256 {
-    value / fast_pow(2.into(), shift.into())
+pub impl BitshiftImpl<
```
Comment (maciejka) [inline src/utils.cairo:9]: Not sure if the trait is necessary. Why not keep it as a generic function?

Comment (TAdev0) [reply src/utils.cairo:9]: Thought It would be more elegant to have a trait for bitshift operation
I can remove it

Comment (maciejka) [reply src/utils.cairo:9]: I don't think it is necessary.

Comment (TAdev0) [reply src/utils.cairo]: @maciejka  just updated the pr

**Pull request merged**